src/sentry/issues/issue_velocity.py

```diff
@@ -133,42 +134,59 @@ def calculate_threshold(project: Project) -> Optional[float]:
     except Exception:
         logger.exception(
-            "Unexpected shape for threshold query results",
-            extra={"project_id": project.id, "result": json.dumps(result)},
+            "sentry.issues.issue_velocity.calculate_threshold.error",
+            extra={"project_id": project.id},
         )
         return None
 
+    return result[0][THRESHOLD_QUANTILE["name"]]
+
 
-def update_threshold(project: Project, threshold_key: str, stale_date_key: str) -> float:
+def update_threshold(
+    project: Project,
+    threshold_key: str,
+    stale_date_key: str,
+    stale_threshold: Optional[float] = None,
+) -> float:
     """
     Runs the calculation for the threshold and saves it and the date it is last updated to Redis.
-    If the threshold is NaN, we save it as 0 at the normal TTL. If the threshold is None (due to errors
-    in the query or any other reason), we save it as 0 at the shortened TTL.
+    If the threshold is NaN, we save it as 0 at the normal TTL. If the threshold is `None` (due to
+    Snuba errors), we save it as the stale threshold if we have one with its existing TTL and
+    update its stale date to make it usable for a set period of time (`FALLBACK_TTL`).
+    Otherwise if we don't have a stale threshold, we save it as 0 with the fallback TTL.
     """
     threshold = calculate_threshold(project)
-    ttl = DEFAULT_TTL
-    if threshold is None:
-        logger.error(
-            "Velocity threshold couldn't be calculated, query returned nothing",
-            extra={"project_id": project.id},
-        )
-        threshold = 0
-        ttl = NONE_TTL
-    elif math.isnan(threshold):  # indicates there were no valid events to base the calculation
+    if threshold is not None and math.isnan(threshold):
         threshold = 0
 
+    ttl = DEFAULT_TTL
+    stale_date = datetime.utcnow()
     client = get_redis_client()
     with client.pipeline() as p:
-        p.set(threshold_key, threshold, ex=ttl)
-        p.set(stale_date_key, convert_date_to_int(datetime.utcnow()), ex=ttl),
+        if threshold is None:  # Snuba failmode
+            # current datetime - the amount of time a threshold is valid for + how much time to wait before trying to query Snuba for the threshold again
+            stale_date = (
+                stale_date
+                - timedelta(seconds=TIME_TO_USE_EXISTING_THRESHOLD)
+                + timedelta(seconds=FALLBACK_TTL)
+            )
+            if stale_threshold is None:
+                ttl = FALLBACK_TTL
+                threshold = 0
+                p.set(threshold_key, threshold, ex=ttl)
+            else:
+                # use the stale threshold and maintain its ttl so threshold and stale date expire in redis at the same time
+                threshold = stale_threshold
+                ttl = client.ttl(threshold_key)
+            metrics.incr("issues.update_new_escalation_threshold", tags={"useFallback": True})
```

Comment:
I feel this method might be more readable if you'd pull `if threshold is None` path into its own method
something like (need better name for the factored out method)

```
def update_threshold(...) -> float:
    threshold = calculate_threshold(project)
    if threshold is None:
        return _get_stale_threshold_snuba_fail_mode(...)
    if math.isnan(threshold):   
         threshold = 0
    client = get_redis_client()
    with client.pipeline() as p:
        p.set(threshold_key, threshold, ex=ttl)
        p.set(stale_date_key, convert_date_to_int(datetime.utcnow()), ex=ttl),
    p.execute()
    return threshold
 
 def _get_stale_threshold_snuba_fail_mode(...) -> float:  
    # current datetime - the amount of time a threshold is valid for + how much time to wait before trying to query Snuba for the threshold again
    stale_date = (
        stale_date
        - timedelta(seconds=TIME_TO_USE_EXISTING_THRESHOLD)
        + timedelta(seconds=FALLBACK_TTL)
    )
    if stale_threshold is None:
        ttl = FALLBACK_TTL
        threshold = 0
        p.set(threshold_key, threshold, ex=ttl)
    else:
        # use the stale threshold and maintain its ttl so threshold and stale date expire in redis at the same time
        threshold = stale_threshold
        ttl = client.ttl(threshold_key)
    metrics.incr("issues.update_new_escalation_threshold", tags={"useFallback": True})
    p.set(stale_date_key, str(stale_date), ex=ttl)
    p.execute()

    return threshold
 ```